### ISSUE TITLE
Android: menu icon overwrites first tab item text

### ISSUE THREAD

**Issue by midu-SA** [fyne-io/fyne#2141]
### Is your feature request related to a problem? Please describe:
On Android (at least my phone) the Menu Icon and the tab bar is located on top of the screen at the same height. Both start on the left side. As a consequence the Menu Icon overwrites the first characters of the first tab item text. See figure.

### Is it possible to construct a solution with the existing API?
Not ideal, but you can use space characters to move the text of the tab item more to the right. In my figure the following text would help:  "     Tasks"

### Describe the solution you'd like to see:
If this effect is identical on all android phones, the tab bar should be placed right of the menu item.

![Screenshot_20210405-201347](https://user-images.githubusercontent.com/66474515/113656750-ce970380-969c-11eb-920d-23232eb762fb.jpg)

**Comment by andydotxyz** [comment]
Good point, this bugs me too.
I think we will need to add some smarter "avoid this spot" logic, because tabs won't always be at the top.
Or we could make a `AppSkeleton` type that lays it all out approprately?

**Comment by stuartmscott** [comment]
Agreed, this should be fixed. Though it isn't specific to tabs - the issue can be replicated in fyne_demo where the menu button is drawn over the first tree node.

Perhaps mobile should have an App Bar that extends the full width of the display containing the menu button, app name, and any actions (https://material.io/components/app-bars-top)

**Comment by fpabl0** [comment]
> we could make a AppSkeleton type that lays it all out approprately?

I like this idea. `AppSkeleton` would be a great addition :)
